### DISCUSSION TITLE
How can I get typing for MiniTest with lua LSP?

### DISCUSSION THREAD

**Discussion by oysandvik94** [nvim-mini/mini.nvim#1357] [Q&A] [answered]
For example, see screenshot:

![image](https://github.com/user-attachments/assets/204190f0-38cb-4e6a-95d4-5256d1991a49)

I would love to get type information on this.

I have installed mini.test in my Neovim config, and I'm using lazydev.nvim.

**Comment by oysandvik94** [comment]
This just started working, maybe I managed to build something correctly :man_shrugging:

**Comment by echasnovski** [comment, marked as answer]
I am afraid, the answer is that there is no intentional LuaLS class definitions coming directly from 'mini.nvim'. See #1110 and #1330 for more details.

If you want to see all `MiniTest` methods, then I guess it is up to LuaLS to infer them correctly (which it might be able to do on practice).

If you want the information about `MiniTest.new_set()` output, then the answer is [basically a table](https://github.com/echasnovski/mini.nvim/blob/ccfc6300a5c17387f2899f979ecc04f942bee2a0/doc/mini-test.txt#L214) without special methods or fields. The common approach in 'mini.nvim' is to try and use as simple data structures as possible (which far less common than I'd like it to be).

**Reply by oysandvik94** [reply]
Thanks for the quick answer!
I am getting typing and autocomplete on methods now, I'm not sure what exactly fixed it, I was installing and running stuff left and right.
